### PR TITLE
Add proposed-work intake triage report

### PR DESCRIPTION
**Pull request by attaboy11**
## Summary

Bounty #694
Source proposed-work intake: #672

Adds a focused read-only proposed-work intake triage report for maintainers:

- `scripts/proposed_work_triage.py` supports offline JSON fixtures and live read-only `gh issue list/view` mode.
- Output can be Markdown or JSON.
- The report detects missing `proposed-work` labels, missing template sections, vague/rejected/already-routed/non-live-confused proposals, and duplicate-looking related proposal groups with evidence tokens and suggested consolidated scopes.
- Live mode also reads public MergeWork API bounty #649 data so accepted proposed-work intake remains separated into pending payout proposals vs proof-backed paid awards.
- Runbook docs explain usage and the read-only boundary.

## Verification

- `PYTHONPATH=. ../mergework/.venv/bin/python -m pytest tests/test_proposed_work_triage.py -q` -> `9 passed`
- `../mergework/.venv/bin/ruff check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py docs/admin-runbook.md` -> `All checks passed!`
- `../mergework/.venv/bin/ruff format --check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py` -> `2 files already formatted`
- `PYTHONPATH=. ../mergework/.venv/bin/python scripts/docs_smoke.py` -> `docs smoke ok`
- `git diff --check origin/main...HEAD` -> passed
- `git merge-tree --write-tree origin/main HEAD` -> clean tree `330be44236be542e2dc3f07f403b1ae29400e203`

Live read-only smoke:

- `PYTHONPATH=. ../mergework/.venv/bin/python scripts/proposed_work_triage.py --repo ramimbo/mergework --format json --limit 10`
- Summary from the smoke: `10` proposed-work issues, payment counts `none=8`, `pending=2`, warning counts included `accepted_pending_payout=2`, `missing_proposed_work_label=1`, `missing_template_sections=1`, `already_routed_or_accepted=5`.
- The section-heading hardening keeps current template labels such as `Possible acceptance criteria`, `Evidence or tests required`, and `Tests` from producing false missing-section warnings.

## Duplicate Check

- Bounty #100 is live/open with `400 MRWK`, `effective_awards_remaining=1`, and no active attempts at submission time.
- Existing PR #720 is the narrower #684/#647 proposed-work fallback report for unlabeled CLI/API-created intake. This PR targets #694/#672's broader intake triage scope, including #649 pending-vs-paid status, duplicate-looking related groups, suggested consolidated scopes, and the accepted #694 test matrix.
- Existing PR #723 changes the proposed-work template source-context field only; it does not add the triage report.

## Out of Scope

No automatic labels, comments, issue edits, bounty creation, claim submission, acceptance decisions, payments, admin-token APIs, production database access, private state, secrets, wallet behavior, price/exchange/bridge/cash-out claims, or private security details.

## Summary by CodeRabbit

* **Documentation**
  * Added an admin runbook subsection "Proposed Work Triage" describing usage, modes, outputs, API host option, and read-only behavior.

* **New Features**
  * Added a triage tool that analyzes proposed-work intake, flags missing template sections/labels, vagueness/rejections, related/duplicate proposals, and payment state; produces JSON or Markdown; supports live GitHub or offline fixture modes and reports data-warnings for incomplete payment info.

* **Tests**
  * Added comprehensive tests covering analysis, grouping, payment-state handling, CLI output, and live-mode read-only/error scenarios.

### CODE REVIEW

File: scripts/proposed_work_triage.py [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+
+
+def _has_section(body: str, aliases: tuple[str, ...]) -> bool:
+    lowered = body.lower()
+    for alias in aliases:
+        if re.search(rf"(^|\n)\s*#+\s*{re.escape(alias)}\b", lowered):
+            return True
+        if re.search(rf"(^|\n)\s*(?:-\s*)?\*\*{re.escape(alias)}\*\*", lowered):
+            return True
+    return False
```
[the remark this answers is not in the document]
Comment (attaboy11) [reply scripts/proposed_work_triage.py:106]: Addressed in current head `d83cf74`: the section aliases now cover `Possible acceptance criteria`, `Evidence or tests required`, and `Tests`, with regression coverage for current template labels and short `Tests` headings. Verification on the updated head: `pytest tests/test_proposed_work_triage.py -q` -> `9 passed`; Ruff check/format, docs smoke, `git diff --check`, and hosted CI are green.

**Comment by attaboy11** [comment]
Addressed the remaining CodeRabbit CLI robustness follow-up in `b2d9101`.

Changes:
- `_run_gh()` now wraps `subprocess.TimeoutExpired` with a clear RuntimeError.
- Invalid/non-JSON `gh` stdout now reports a concise invalid-JSON RuntimeError with an output excerpt.
- Added regression coverage for timeout and invalid JSON handling.

Validation:
- `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 ../mergework-pr757-review/.venv/bin/python -m pytest tests/test_proposed_work_triage.py -q` -> 10 passed.
- `../mergework-pr757-review/.venv/bin/python -m ruff check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py docs/admin-runbook.md` -> passed.
- `git diff --check origin/main...HEAD` -> passed.

**Comment by MyTH-zyxeon** [comment]
Current-head review for PR #763 (`b2d91016aeb349b8ed269760e3b3cc9f30e35d55`).

I reviewed the current diff as a non-author for the #654 review bounty. The PR is open with `mergeStateStatus=CLEAN`; hosted Quality/readiness/docs/image checks pass; CodeRabbit is marked pass/review skipped.

What I checked:

- `docs/admin-runbook.md` adds a proposed-work intake triage workflow and clearly warns that proposed-work issues are not live claimable bounties.
- `scripts/proposed_work_triage.py` supports offline JSON input and live read-only collection through hardcoded `gh issue list` / `gh issue view` calls.
- The analyzer flags missing `proposed-work` labels, missing template sections, vague intake, rejected/routed proposals, non-live bounty confusion, likely related proposal groups, and payment status from pending payout proposals or proof-backed awards.
- Live mode reads public MergeWork bounty #649 state and keeps pending payout proposals distinct from proof-backed paid awards.
- Tests cover complete and incomplete proposals, current template section labels, short tests headings, conservative related grouping, pending-vs-paid payment status, rejected/non-live-confused proposals, CLI markdown/json output, read-only live mode calls, and timeout/invalid JSON errors.

No blocking issue found in this review. The PR is useful as a focused intake triage tool and the implementation stays within the read-only boundary currently exercised by live mode.

Useful follow-up before depending on the live report for payment reconciliation: `_load_public_bounty_issue()` catches public API/network/JSON failures and returns an empty list. That avoids breaking the report, but it can silently change `accepted_pending_payout` or `proof_backed_paid` rows back to `none` when the public API is temporarily unavailable or its shape changes. Consider adding a visible warning in the report summary when the public payment-state fetch fails, or at least include a test for that downgrade path so maintainers know the payment section is incomplete rather than empty.

Payout-routing note: the PR body references #694, but this review is for the #654 review bounty only; it does not assert that #694 itself is currently claimable or payable.

**Comment by attaboy11** [comment]
Pushed follow-up head `46e460c` addressing the current-head review note about silent payment-state downgrades.

Changes:
- live-mode public bounty/payment enrichment now returns `summary.data_warnings` when the bounty list fetch fails.
- bounty detail fetch failures now add a `payment_state_incomplete` warning and fall back to the list row instead of silently treating the payment section as complete.
- Markdown output renders these as `data warning:` summary lines.
- Added a live-mode regression where the public MRWK API is unavailable; the report keeps `payment_counts` honest and exposes the incomplete payment-state warning.

Validation:
- `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 ... python -m pytest tests/test_proposed_work_triage.py -q` -> 11 passed.
- `python -m pytest tests/test_proposed_work_triage.py tests/test_pr_queue_health.py tests/test_claim_inventory.py -q` -> 35 passed.
- `python -m pytest -q` -> 634 passed, 1 warning.
- `python -m ruff check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py docs/admin-runbook.md` -> passed.
- `python -m ruff format --check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py` -> 2 files already formatted.
- `git diff --check origin/main...HEAD` and `git diff --check` -> passed.
- `git merge-tree --write-tree origin/main HEAD` -> clean tree `e1c727840c52a061298598def5f1bb7e5dad51b3`.
- live read-only smoke: `scripts/proposed_work_triage.py --repo ramimbo/mergework --limit 3 --format json` -> 3 issues, payment counts `{"none": 2, "pending": 1}`, `data_warnings: []`.
- hosted Quality/readiness/docs/image check on head `46e460c`: passed.

CodeRabbit is still processing the latest head as of this comment; I will watch for actionable follow-up.

File: tests/test_proposed_work_triage.py [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+    assert main(["--repo", "ramimbo/mergework", "--format", "json", "--limit", "1"]) == 0
+    output = json.loads(capsys.readouterr().out)
+
+    assert output["summary"]["payment_counts"] == {"none": 1}
+    assert output["summary"]["data_warnings"] == [
+        "payment_state_incomplete: failed to load public bounty list for issue #649 (URLError)"
+    ]
+    markdown = format_markdown(output)
+    assert "data warning: payment_state_incomplete" in markdown
+
```
[the remark this answers is not in the document]
Comment (attaboy11) [reply tests/test_proposed_work_triage.py:424]: Addressed in `ca89183`: hoisted `FakeResponse` to module scope and added `test_proposed_work_triage_live_mode_warns_when_bounty_detail_fetch_fails`, covering the list-succeeds/detail-fetch-fails payment warning path. Validation: `tests/test_proposed_work_triage.py` -> 12 passed; focused triage/queue/claim tests -> 36 passed; full suite -> 635 passed, 1 warning; Ruff check/format and diff checks passed; hosted CI passed on `ca89183`.

**Comment by attaboy11** [comment]
Follow-up pushed in `ca89183` for CodeRabbit's detail-fetch downgrade test note.

Changes:
- Hoisted the live-mode `FakeResponse` helper to module scope.
- Added `test_proposed_work_triage_live_mode_warns_when_bounty_detail_fetch_fails` to cover the branch where the bounty list loads but a per-bounty detail request fails.
- The test asserts the visible `payment_state_incomplete` warning and Markdown data-warning output.

Validation:
- `.venv/bin/python -m pytest tests/test_proposed_work_triage.py -q` -> 12 passed.
- `.venv/bin/python -m pytest tests/test_proposed_work_triage.py tests/test_pr_queue_health.py tests/test_claim_inventory.py -q` -> 36 passed.
- `.venv/bin/python -m pytest -q` -> 635 passed, 1 warning.
- `.venv/bin/ruff check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py` -> passed.
- `.venv/bin/ruff format --check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py` -> 2 files already formatted.
- `git diff --check` and `git diff --check origin/main...HEAD` -> passed.
- Hosted Quality/readiness/docs/image check on `ca89183`: passed.
- CodeRabbit on `ca89183`: passed / review skipped.

CodeRabbit marked the inline comment addressed in `ca89183`, and both required status contexts are now green.

**Review by aunysillyme — COMMENTED** [review]
### PR #763 Triage Report Review (Bounty #694)

Detailed comment review for PR #763 (`Add proposed-work intake triage report`) by `attaboy11`.

#### Evidence:
- **Commit Head Inspected:** `ca891839cbcc95ef20160fb90db0541beb582668` on branch `pr-763`.
- **Files Inspected:** `scripts/proposed_work_triage.py`, `tests/test_proposed_work_triage.py`, and `docs/admin-runbook.md`.
- **Verification Checked:** Inspected the API requests and live collection logic.
- **Actionable Findings / Structural comparison:**
  - In `load_live_issues`, the script makes separate HTTP requests to the MergeWork API for each specific issue number in a loop to query the bounty details. This loop-based design is prone to network bottlenecks and rate limits when scanning a large number of issues, compared to the single-pass state-load approach of PR #766.
  - The script does not implement duplicate clustering or token similarity checks to help maintainers detect related proposal scopes.
  - The safety guards for the `gh` commands are basic, lacking PR #766's thorough tokenizer check that actively rejects mutating parameters and non-read-only GET/HEAD methods.

The PR is a solid starting implementation but is less performant on live network operations and lacks key security guards and duplicate detection features compared to PR #766.

**Comment by attaboy11** [comment]
Follow-up pushed in `5cbbb3d11dfe124d980793bfb395e540afa0c3a2` to strengthen the live-mode GitHub safety boundary.

Changes:
- Added an explicit allowlist around `_run_gh()`: live mode now permits only `gh issue list` and `gh issue view`.
- Added regression coverage proving mutating commands such as `gh issue comment`, `gh issue edit`, `gh pr review`, and `gh api ...` are rejected before `subprocess.run` is called.

Clarifications on the structural-comparison review:
- Duplicate/related proposal grouping is implemented by `_related_groups()` plus `_mark_duplicate_warnings()`, and covered by `test_proposed_work_triage_groups_likely_related_proposals_conservatively`.
- Live public MRWK API detail fetches are bounded to the small #649 bounty lookup rows, not one request per proposed-work issue being scanned. Issue collection itself remains `gh issue list` / `gh issue view` only.

Validation on current head:
- `pytest tests/test_proposed_work_triage.py -q` -> 13 passed
- `pytest tests/test_proposed_work_triage.py tests/test_pr_queue_health.py tests/test_claim_inventory.py -q` -> 37 passed
- full suite `pytest -q` -> 636 passed, 1 warning
- `ruff check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py` -> passed
- `ruff format --check scripts/proposed_work_triage.py tests/test_proposed_work_triage.py` -> 2 files already formatted
- `git diff --check` -> passed
- live read-only smoke `scripts/proposed_work_triage.py --repo ramimbo/mergework --limit 3 --format json` -> 3 issues, payment counts `{"none": 2, "pending": 1}`, `data_warnings: []`
- hosted CI: passed
- CodeRabbit: success
- merge state: clean

**Review by JONASXZB — APPROVED** [review]
APPROVE — reviewed current head `5cbbb3d11dfe124d980793bfb395e540afa0c3a2` as a non-author.

Evidence checked:
- Inspected the current diff in `scripts/proposed_work_triage.py`, `tests/test_proposed_work_triage.py`, and `docs/admin-runbook.md`.
- Verified the latest head's `_run_gh()` hard allowlist only permits `gh issue list` and `gh issue view`; mutating examples such as `gh issue comment`, `gh issue edit`, `gh pr review`, and `gh api ...` are rejected before `subprocess.run` by `test_run_gh_rejects_mutating_commands_before_subprocess`.
- Checked live-mode public payment-state handling: bounty list fetch failures and per-bounty detail failures now produce `payment_state_incomplete` data warnings instead of silently downgrading pending/paid state to `none`.
- Checked offline fixture support through `--input`, JSON/Markdown formatting, missing label/template detection, vague/rejected/non-live-confusion warnings, conservative related-proposal grouping, and #649 pending-vs-proof-backed payment classification in the focused tests.
- Local validation on this head: `pytest tests/test_proposed_work_triage.py -q` -> 13 passed; targeted `ruff check` passed; targeted `ruff format --check` passed.
- Mergeability/hosted status: `merge-tree` against current `origin/main` produced clean tree `00fe17fafa48761da5677d4be85d7810b91f799c`; GitHub mergeable state is clean; combined status and hosted Quality/readiness/docs/image check are success.

No blocker found in the current head. Minor non-blocking cleanup: `_related_groups()` has a duplicated unreachable `return sorted(...)` line, but it does not affect behavior or the bounty acceptance surface.
